src/fs_event_wrap.cc

```diff
@@ -156,7 +160,31 @@ void FSEventWrap::OnEvent(uv_fs_event_t* handle, const char* filename,
   };
 
   if (filename != nullptr) {
-    argv[2] = OneByteString(env->isolate(), filename);
+    /* If encoding is BUFFER, return a Buffer.
+       If encoding is UCS2, attempt to decode as UCS2 or report an error.
+       Otherwise, Encode as the requested type, with UTF8 as the fallback. */
+    if (wrap->encoding_ == BUFFER) {
+      argv[2] = Buffer::Copy(env,
+                             filename,
+                             strlen(filename)).ToLocalChecked().As<Value>();
+    } else if (wrap->encoding_ == UCS2) {
```

Comment:
Don't you mean `!= UCS2`? Because if `encoding_ == UCS2` then should be making the call to `StringBytes::Encode()` where a `uint16_t*` is required. Which happens in the following `else`.